hasher-matcher-actioner/src/OpenMediaMatch/blueprints/hashing.py

```diff
@@ -72,6 +77,33 @@ def is_valid_url(url: str) -> bool:
         return False
 
 
+def _get_and_check_content_length(
+    url: str, max_length: int = MAX_CONTENT_LENGTH
+) -> requests.Response:
+    """
+    Get URL content with explicit content length tracking.
+    Raises an exception if content length exceeds max_length.
+    """
+    # First check content length with HEAD request
+    head_resp = requests.head(url, timeout=30, allow_redirects=True)
+    head_resp.raise_for_status()
+
+    content_length = head_resp.headers.get("content-length")
+    if content_length is not None and int(content_length) > max_length:
+        abort(413, "Content too large")
+
+    # If content length is acceptable, proceed with GET request
+    response = requests.get(url, stream=True, timeout=30, allow_redirects=True)
+    response.raise_for_status()
+
+    # Double check content length from GET response
+    content_length = response.headers.get("content-length")
+    if content_length is not None and int(content_length) > max_length:
+        abort(413, "Content too large")
```

Comment:
Do you need to do something to explicitly close the connection here?

Reply:
used context manager instead to ensure we clean-up in case of any exception.